### PR TITLE
test: Refactor Advanced Gas fees test to utilize mocks

### PR DESCRIPTION
**Pull request by SamuelSalas**
## **Description**
The advanced gas fees tests are becoming a bit unstable within the smoke pipeline mainly because we are relying on real funds (Sepolia ETH) to submit the transaction. With the recent addition of mocking, we should refactor the advanced gas fees tests to:

- use ganache
- mock the gas api to return expected values

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: e2e/pages/Send/AmountView.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import Matchers from '../../utils/Matchers';
 import Gestures from '../../utils/Gestures';
-import { AmountViewSelectorsIDs } from '../../selectors/SendFlow/AmountView.selectors';
+import {
+  AmountViewSelectorsIDs,
+  AmountViewSelectorsText
+} from '../../selectors/SendFlow/AmountView.selectors';
 
 class AmountView {
-  get currencySwitch() {
```
Comment (cortisiko) [inline e2e/pages/Send/AmountView.js:6]: I recall we had a specific use case for this. Unfortunately, the test that used this test step was deleted https://github.com/MetaMask/metamask-mobile/blob/3635e729262bf8db08983e79f80bba7d67f5df99/e2e/specs/confirmations/send-eth-flow.spec.js#L58 . We can work to add it back. Do not remove.

File: e2e/pages/Send/AmountView.js [hunk 2]
```diff
@@ -26,9 +29,5 @@ class AmountView {
       ? await Gestures.typeTextAndHideKeyboard(this.amountInputField, amount)
       : await Gestures.replaceTextInField(this.amountInputField, amount);
   }
-
-  async tapCurrencySwitch() {
```
Comment (cortisiko) [inline e2e/pages/Send/AmountView.js:30]: I recall we had a specific use case for this. Unfortunately, the test that used this test step was deleted https://github.com/MetaMask/metamask-mobile/blob/3635e729262bf8db08983e79f80bba7d67f5df99/e2e/specs/confirmations/send-eth-flow.spec.js#L58 . We can work to add it back. Do not remove.

**Review by cortisiko — CHANGES_REQUESTED** [review]
nice stuff! small feedback

File: e2e/pages/Send/AmountView.js (hunk 1 above)
Comment (SamuelSalas) [reply e2e/pages/Send/AmountView.js:6]: Done

File: e2e/pages/Send/AmountView.js (hunk 2 above)
Comment (SamuelSalas) [reply e2e/pages/Send/AmountView.js]: Done

**Review by cortisiko — APPROVED** [review]

**Pull request merged**